### DISCUSSION TITLE
How do I make a workflow produce an artifact?

### DISCUSSION THREAD

**Discussion by skye** [community/community#25536] [Actions]
<p>I’m trying to set up a build workflow for that produces several artifacts via a script. I’ve got as far as successfully running the build scripts via the workflow, which the logs indicate are producing the artifacts as expected, but I don’t see the artifact menu as described in <a href="https://help.github.com/en/articles/managing-a-workflow-run#downloading-logs-and-artifacts" rel="nofollow noopener">https://help.github.com/en/articles/managing-a-workflow-run#downloading-logs-and-artifacts</a>. Do I have to explicitly mark the files to be archived somehow?</p>

**Comment by rajsite** [comment, marked as answer]
<p>I think to make artifacts show up there you can use GitHub’s upload-artifact action: <a href="https://github.community/t/caching-files-between-github-action-executions/16047/5">https://github.community/t5/GitHub-Actions/Caching-files-between-GitHub-Action-executions/m-p/30172/highlight/true#M130</a></p>
<p>This is the actual upload-artifact action repo: <a href="https://github.com/actions/upload-artifact" rel="nofollow noopener">https://github.com/actions/upload-artifact</a></p>

**Comment by joebowbeer** [comment]
<p>More here:</p>
<p><a href="https://help.github.com/en/articles/persisting-workflow-data-using-artifacts" rel="nofollow noopener">https://help.github.com/en/articles/persisting-workflow-data-using-artifacts</a></p>

**Comment by bejado** [comment]
<p>Is there a way to manually upload an artifact, without using GitHub’s upload-artifact action? Our organization has enabled the “Enable local actions only” restriction so we’re unable to use actions outside of our own repository.</p>

**Comment by Krijger** [comment]
<p>Isn’t the action open source? In which case, your organisation can simply fork to fix this issue?</p>
